### PR TITLE
Introducing the Ports & Adapters to `aruba`'s codebase

### PR DESCRIPTION
**Pull request by dg-ratiodata**
This is the PR for #298. It introduces three classes to aruba, which abstract the os specific code. @mattwynne Matt, is this what you've in mind? Pinging @weedySeaDragon as well.

I split this PR in several commits, each one can be reviewed by it's own.

### CODE REVIEW

**Comment by dg-ratiodata** [comment]
BTW: The jruby-failures seem to be somehow related to `0.8.0`... I hope to fix them soon.

File: lib/aruba/platform.rb [hunk 1]
```diff
@@ -1,257 +1,29 @@
...
+      @platforms = []
+      @platforms << Aruba::Platforms::WindowsPlatform
+      @platforms << Aruba::Platforms::UnixPlatform
 
-    # Transform hash to a string table which can be output on stderr/stdout
-    def simple_table(hash)
-      SimpleTable.new(hash).to_s
+      super @platforms.find(&:match?).new
     end
+  end
```
Comment (mattwynne) [inline lib/aruba/platform.rb]: The other way to do this is simply to do:

```
module Aruba
  Platform = [Platforms::WindowsPlatform, Platforms::UnixPlatform].find(&:match).new
end
```

File: lib/aruba/platform.rb [hunk 2]
```diff
@@ -1,257 +1,29 @@
...
-    def write_file(path, content)
-      if RUBY_VERSION < '1.9.3'
-        File.open(path, 'wb') do |f|
-          f.print content
-        end
-      else
-        File.write(path, content)
-      end
-    end
+  PLATFORM_MUTEX = Mutex.new
```
Comment (mattwynne) [inline lib/aruba/platform.rb]: What's this for?

**Comment by mattwynne** [comment]
:claps:

File: lib/aruba/platform.rb (hunk 2 above)
Comment (dg-ratiodata) [inline lib/aruba/platform.rb]: Paranoia... I read something about class instance variables being not thread safe.

File: lib/aruba/platform.rb (hunk 1 above)
Comment (dg-ratiodata) [inline lib/aruba/platform.rb]: That's a pretty one. Fixed in master.

File: lib/aruba/platform.rb (hunk 2 above)
Comment (mattwynne) [inline lib/aruba/platform.rb]: That's only if you mutate things. We're keeping this as a constant now, so it should be fine. Let's just solve the problems we have in front of us! :)

**Pull request merged**